### ISSUE TITLE
wrong precedence of JUPYTER_PATH env variable

### ISSUE THREAD

**Issue by haraldschilly** [nteract/kernelspecs#28]
The path(s) in the `JUPYTER_PATH` env variable should take precedence over the other default jupyter path variables. see

> Directories given in JUPYTER_PATH are searched before other locations.

http://jupyter.readthedocs.io/en/latest/projects/jupyter-directories.html#envvar-JUPYTER_PATH

However, I have here two `python3` kernel directories, one in my `$JUPYTER_PATH/kernels/...` directory and another one is a system-wide `/usr/local/share/jupyter/kernels/python3`.

Unfortunately, `kernelspecs` returns  `python3: { ... resources_dir: '/usr/local/share/jupyter/kernels/python3'}` as the python3 kernel, and not my other customized python3 kernel.

**Comment by haraldschilly** [comment]
Thinking aloud while poking around in the code, this happens here running `jupyter-path` directly:

```
coffee> jp.dataDirs(withSysPrefix: true).then(console.log)
Promise { <pending> }
coffee> [ '/ext/jupyter',         <- my $JUPYTER_PATH
  '/home/user/.local/share/jupyter',
  '/usr/local/share/jupyter',
  '/usr/share/jupyter' ]
```

**Comment by haraldschilly** [comment]
I think the problem is in traverse.js's line `kernels[kernel.name] = kernel;`. It overwrites an already existing definition with the lower-precedence definition. Not sure, if its ok to put an "if" there or does some functional/reduce operator construction exist to do this?

**Comment by rgbkrk** [comment]
I think the if is just fine. We'd need to reverse the array otherwise.

**Comment by rgbkrk** [comment]
Closed by #29.
